scipy/optimize/__lbfgsb.c

```diff
@@ -1761,8 +1736,6 @@ cauchy(int n, double* x, double* l, double* u,
             // to repeat the loop for unsearched intervals.
             dtm = -f1 / f2;
         } else if (bnded) {
-            f1 = 0.0;
-            f2 = 0.0;
```

Comment:
Why did you choose to remove these lines? Not sure if they do anything but they are in the source. 

https://github.com/scipy/scipy/blob/adc4f4f7bab120ccfab9383aba272954a0a12fb0/scipy/optimize/lbfgsb_src/lbfgsb.f#L1653-L1663

Reply:
If I'm not mistaken, those updates are for the logging below (https://github.com/scipy/scipy/blob/adc4f4f7bab120ccfab9383aba272954a0a12fb0/scipy%2Foptimize%2Flbfgsb_src%2Flbfgsb.f#L1671) in the reference implementation. Actually clang-tidy detected those lines as "dead stores".

If in doubt, I can bring back those lines.

Reply:
I don't really have any preference actually. Since these are `break` branches, I think it is safe to leave out.